From bbbad0afef910ba70a56da89389e121fb7cec2e0 Mon Sep 17 00:00:00 2001
From: Konstantin Khlebnikov <khlebnikov@openvz.org>
Date: Wed, 3 Jun 2009 16:33:00 +0400
Subject: [PATCH 86/86] pidns: pi-futex pid check fixup

fix WARN_ON condition

port rh5 patch from Stanichenko Marat <mstanichenko@openvz.org>

http://bugzilla.openvz.org/show_bug.cgi?id=1262
---
 kernel/futex.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/kernel/futex.c b/kernel/futex.c
index a02be16..ed2c6ed 100644
--- a/kernel/futex.c
+++ b/kernel/futex.c
@@ -534,7 +534,7 @@ lookup_pi_state(u32 uval, struct futex_hash_bucket *hb,
 
 			WARN_ON(!atomic_read(&pi_state->refcount));
 			WARN_ON(pid && pi_state->owner &&
-				pi_state->owner->pid != pid);
+				task_pid_vnr(pi_state->owner) != pid);
 
 			atomic_inc(&pi_state->refcount);
 			*ps = pi_state;
-- 
1.6.3.1

